extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusWorkerPoolRegistry.java

```diff
@@ -30,14 +40,73 @@
 @ApplicationScoped
 // TODO: create a different entry for WorkerPoolRegistry than `analyzeWorker` and drop this class
 public class QuarkusWorkerPoolRegistry extends WorkerPoolRegistry {
+
+    private static final Logger logger = Logger.getLogger(QuarkusWorkerPoolRegistry.class);
     private static final String WORKER_CONFIG_PREFIX = "smallrye.messaging.worker";
     private static final String WORKER_CONCURRENCY = "max-concurrency";
+    public static final String DEFAULT_VIRTUAL_THREAD_WORKER = "<virtual-thread>";
 
     @Inject
     ExecutionHolder executionHolder;
 
     private final Map<String, Integer> workerConcurrency = new HashMap<>();
     private final Map<String, WorkerExecutor> workerExecutors = new ConcurrentHashMap<>();
+    private final Set<String> virtualThreadWorkers = initVirtualThreadWorkers();
+
+    private static Set<String> initVirtualThreadWorkers() {
+        Set<String> set = new ConcurrentHashSet<>();
+        set.add(DEFAULT_VIRTUAL_THREAD_WORKER);
+        return set;
+    }
+
+    public static final Supplier<Executor> VIRTUAL_EXECUTOR_SUPPLIER = new Supplier<Executor>() {
+        Executor current = null;
```

Comment:
This could use a double-checked locking pattern here, if possible - or, even better a 

```java
private enum VirtualExecutorSupplier implements Supplier<Executor> {
     Instance;
     
     private final Executor executor;
     
     VirtualExecutorSupplier() {
          // bla bla bla long initialization
     }
     
     @Override
     public Executor get() {
            return executor;  
     }
}
```

Reply:
I didn't knew the pattern with an `enum` 😉

Reply:
It saves thinking about mutability of fields and give for free the lazines property (although introduce a new class for it)

Reply:
Yes, the only thing is the enum init would be captured during native image compilation.

Reply:
Configuring the enum as runtime initialized did the trick for native image.